src/perf_ringbuffer.c

```diff
@@ -245,12 +250,19 @@ perf_event_sample *hdr2samp(const struct perf_event_header *hdr,
     sample.ips = buf;
     buf += sample.nr;
   }
-  if (PERF_SAMPLE_RAW & mask) {}
+  if (PERF_SAMPLE_RAW & mask) {
+    // size_raw is a 32-bit integer!
+    sample.size_raw = ((flipper *)buf)->half[0];
+    sample.data_raw = sample.size_raw ? &((flipper *)buf)->half[1] : NULL;
+    buf += 1 + (sample.size_raw / sizeof(*buf)); // Advance + align
+  }
   if (PERF_SAMPLE_BRANCH_STACK & mask) {}
   if (PERF_SAMPLE_REGS_USER & mask) {
     sample.abi = *buf++;
-    // In case regs are not available, ignore this sample
-    if (sample.abi == PERF_SAMPLE_REGS_ABI_NONE) {
+    // Since some of the previous events are variable-length, ABI gives us
+    // an opportunity to sanity-check the buffer (only 2 legal values!)
+    if (sample.abi != PERF_SAMPLE_REGS_ABI_32 &&
```

Comment:
Not sure to follow the reasoning here, isn't `PERF_SAMPLE_REGS_ABI_NONE` a legal value ?

Reply:
`PERF_SAMPLE_REGS_ABI_NONE` is legal, though we will not do anything with it. 
The reasoning is to make the check less permissive if somehow we miss-align reads, we won't create a bad sample / crash in the following lines.

Reply:
[Minor] The comment is a bit misleading then
It could be interesting to separate `PERF_SAMPLE_REGS_ABI_NONE` case (we currently return NULL, but from the ring buffer perspective, sample is valid) and other invalid values.